### PR TITLE
feat: mf-3417 wipe out fake identity

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-3417

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/EVM/src/state/IdentityService.ts [hunk 1]
```diff
@@ -284,9 +284,31 @@ export class IdentityService extends IdentityServiceState<ChainId> {
...
+
+        const identities = uniqBy(identities_, (x) => [x.type, x.label, x.address.toLowerCase()].join('_'))
+
+        if (!getNextIDPlatform() && !(process.env.channel === 'stable' && process.env.NODE_ENV === 'production')) {
+            return identities
+        }
+
+        const allSettledTwitterHandler = await Promise.allSettled(
+            identities.map(async (x) => {
+                const bindingProofs = await NextIDProof.queryProfilesByRelationService(x.address)
```
Comment (guanbinrui) [inline packages/plugins/EVM/src/state/IdentityService.ts]: The `identities` could have duplicate addresses.

Comment (guanbinrui) [reply packages/plugins/EVM/src/state/IdentityService.ts]: Can we query it like this?

```ts
queryProfilesByRelationService(uniqBy(identities.map(x => x.address), x => x.toLowerCase()))
```

Comment (hanszhoou) [reply packages/plugins/EVM/src/state/IdentityService.ts]: > The `identities` could have duplicate addresses.

fixed.


> Can we query it like this?
> 
> ```ts
> queryProfilesByRelationService(uniqBy(identities.map(x => x.address), x => x.toLowerCase()))
> ```

No, backend limitation.

Comment (guanbinrui) [reply packages/plugins/EVM/src/state/IdentityService.ts]: I see the requirement only mentioned `ENS` and `Lens`. So detecting `SocialAddressType` is also helpful. However, some of them might not need to verify by API.

Comment (hanszhoou) [reply packages/plugins/EVM/src/state/IdentityService.ts]: Address is also required.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**